ement-room.el

```diff
@@ -2787,135 +2800,118 @@ ROOM defaults to the value of `ement-room'."
 
 ;; ;; ;; URLs
 
-(defvar-local ement-room--preview-stop nil
-  "Whether to proceed to previews for current room or not. Used in case of rate limit.")
-
-(defun ement-room--linkify-urls (string msg)
+(defun ement-room--linkify-urls (string message)
```

Comment:
Again, this function should not be overloaded to do anything regarding previews.  It shouldn't be necessary to change this function in this PR.

Reply:
Basically, yes.  There should probably be a new function named something like `ement-room--preview-urls` that will receive the event as an argument, and when it finds URLs to preview in the event's body, it should call a function for each URL which requests the preview from the server and uses a callback to add the preview data to the event's local slot and invalidate the event's node when the room has a buffer.

We could, instead, fetch the previews when the event is formatted for the first time, so as to not fetch previews for events that are never going to be displayed (i.e. in rooms whose buffers aren't going to be shown).  That's probably better, in fact, but the rest of the procedure remains mostly the same.

I don't think the extra parsing will be a performance problem, but if it is, we can address it then.  We should avoid premature optimization.  :)